validator/staker.go

```diff
@@ -167,13 +170,46 @@ func NewStaker(
 		lastActCalledBlock:  nil,
 		withdrawDestination: withdrawDestination,
 		inboxReader:         inboxReader,
+		nitroMachineConfig:  nitroMachineConfig,
```

Comment:
why keep nitroMachineConfig - and not NitroMachineLoader?
Seems reasonable to have one NitroMachineLoader - use it in both blockValidaror and Here. This NitroMachineLoader could have a GetLatestWasmRoot func.